test/quantum_objects.jl

```diff
@@ -576,6 +589,50 @@
         @test ρ1.data ≈ ρ1_ptr.data atol = 1e-10
         @test ρ2.data ≈ ρ2_ptr.data atol = 1e-10
 
+        ψlist = [rand_ket(2), rand_ket(3), rand_ket(4), rand_ket(5)]
+        ρlist = [rand_dm(2), rand_dm(3), rand_dm(4), rand_dm(5)]
+        ψtotal = tensor(ψlist...)
```

Comment:
All these cases are non entangled states. Could yo add some entanglement? For example testing also `ψtotal_entangled = normalize(tensor(ψlist2…) + 1im * tensor(ψlist2…))` (where `ψlist2`) is another list of random kets. The degree of entanglement is determined by the random kets, but we should have some form of entanglement. Same for the density matrices.